openedx/core/djangoapps/content/block_structure/config/__init__.py

```diff
@@ -10,12 +10,41 @@
 from .models import BlockStructureConfiguration
 
 # Switches
+# .. toggle_name: INVALIDATE_CACHE_ON_PUBLISH
+# .. toggle_implementation: WaffleSwitch
+# .. toggle_default: False
+# .. toggle_description: Invalidates block structure cache when changes to courses are published.
+# .. toggle_use_cases: opt_in
+# .. toggle_creation_date: 2017-02-23
+# .. toggle_tickets: https://github.com/edx/edx-platform/pull/14571
 INVALIDATE_CACHE_ON_PUBLISH = WaffleSwitch(
     "block_structure.invalidate_cache_on_publish", __name__
 )
+
+# .. toggle_name: STORAGE_BACKING_FOR_CACHE
+# .. toggle_implementation: WaffleSwitch
+# .. toggle_default: False
+# .. toggle_description: Enables retrieving block structures from storage instead of regenerating
+#   block structures when they aren't available in cache. 
+# .. toggle_warnings: Depends on the storage class, `STORAGE_CLASS`, and storage keyword arguments,
+#   `STORAGE_KWARGS`, specified under `BLOCK_STRUCTURES_SETTINGS`.
+# .. toggle_use_cases: opt_in
+# .. toggle_creation_date: 2017-02-23
+# .. toggle_tickets: https://github.com/edx/edx-platform/pull/14512
 STORAGE_BACKING_FOR_CACHE = WaffleSwitch(
     "block_structure.storage_backing_for_cache", __name__
 )
+
+# .. toggle_name: RAISE_ERROR_WHEN_NOT_FOUND
+# .. toggle_implementation: WaffleSwitch
+# .. toggle_default: False
+# .. toggle_description: Raises an error if the requested block structure does not exist in block
+#   structure store, or if it is outdated. Block structure store refers to both cache and storage,
+#   if enabled.
+# .. toggle_use_cases: temporary
```

Comment:
I know this sounds weird, since this has been there for 4 years, but there's a [comment](https://github.com/edx/edx-platform/pull/14512/files#r103130819) from @nasthagiri, author of the `RAISE_ERROR_WHEN_NOT_FOUND` change mentioning that the switch is temporary in #14512.

> **Note to reviewers**: We now plan to raise an error instead, if the block-structure is not available when requested. This is now an unexpected situation since the block-structure now has permanent storage. We honor the `RAISE_ERROR_WHEN_NOT_FOUND` waffle switch in the interim as we transition over to the new strategy.